src/base/bittorrent/session.cpp

```diff
@@ -3068,6 +3070,8 @@ void Session::startUpTorrents()
         if (!addTorrent_impl(params.addTorrentData, params.magnetUri, TorrentInfo::loadFromFile(filePath), params.data))
             logger->addMessage(tr("Unable to resume torrent '%1'.", "e.g: Unable to resume torrent 'hash'.")
                                .arg(params.hash), Log::CRITICAL);
+        // don't waste memory with to many unprocessed queue entries
+        readAlerts();
```

Comment:
That's bad, pecause we can't itterate 20 000 torrents without handle the alerts.

If we are not ready to read alerts, then we have to move the whole add torrent call to the point at which we are ready to handle all alert types.

Do you have a type in mind which should not be ignored because of missing gui?

Or we just setup max of torrent count, connectionlimit*10, 1000 and hope that we have enoght memory (I didn't check the alert structure)